src/platform/input/game-pads.js

```diff
@@ -136,109 +734,115 @@ class GamePads {
 
         if (this.gamepadsSupported) {
             const padDevices = navigator.getGamepads ? navigator.getGamepads() : navigator.webkitGetGamepads();
+
             for (let i = 0, len = padDevices.length; i < len; i++) {
                 if (padDevices[i]) {
-                    pads.push({
-                        map: this.getMap(padDevices[i]),
-                        pad: padDevices[i]
-                    });
+                    const pad = this.findByIndex(padDevices[i].index);
+
+                    if (pad) {
+                        pads.push(pad.update(padDevices[i]));
+                    } else {
+                        const nPad = new GamePad(padDevices[i], this.getMap(padDevices[i]));
+                        this.current.push(nPad);
+                        pads.push(nPad);
+                    }
                 }
             }
         }
+
         return pads;
     }
 
+    /**
+     * Destroy the event listeners.
+     *
+     * @ignore
+     */
+    destroy() {
+        window.removeEventListener('gamepadconnected', this._ongamepadconnectedHandler, false);
+        window.removeEventListener('gamepaddisconnected', this._ongamepaddisconnectedHandler, false);
+    }
+
     getMap(pad) {
```

Comment:
Just optional, but I would still like typing here for type safety

Reply:
It's fixed, I used your docs, for it. Hope you won't mind!